### ISSUE TITLE
Lowest percentage of overlap between two point clouds

### ISSUE THREAD

**Issue by WilliamKRobert** [chrischoy/FCGF#3]
In the paper, the percentage of overlap is at least 30%. But I was not able to get the training to converge on my dataset which has 50% overlaps. The input point clouds are from one CAD model, one point cloud keeps all the sampled surface points from CAD, the other one is obtained by first cuts the CAD model into half, and then sampled surface points from half of the model.

**Comment by chrischoy** [comment]
It is 30%. It is unclear what you mean by converge. After few epochs, the FMR will increase, but the loss would stay almost the same as the dimension of the feature limits overfitting to the problem. Did you measure the hit ratio or FMR and saying those don't converge?

**Comment by chrischoy** [comment]
I think it might help if you first follow the training procedure on 3DMatch in README.md and compare how the training progresses.

**Comment by WilliamKRobert** [comment]
I see your point. The hit ratio did increase slowly with oscillation, e.g. it is 0.111 after 50 epochs training but it is growing. I guess it should finally increase to some good values? (I set the voxel size as 1mm and hit_ratio_thresh 4mm)

**Comment by chrischoy** [comment]
Hmm, try different voxel sizes. Usually larger voxel size gives pretty good numbers due to more connectivity. We are using convolution, if there's no neighbor, it will become a PointNet.

Experimentally, when I use 1cm for 3DMatch scenes, the performance decreases. There's a trade-off between registration resolution and receptive field size and sparsity.

On a different note, the Feature Match Ratio hyper parameters widely used in this area are 10cm threshold (hit_ratio_threshold) with 5% inlier ratio (hit ratio) for indoor scenes. The rationale is that with 5% inliers, RANSAC can find the correct registration with 99.9% time after 4M iterations. With 10% hit ratio, it is above the bare minimum 5% although RANSAC would take a lot longer.

**Comment by WilliamKRobert** [comment]
Fair enough. My experiments also gave above 30% hit ratio with voxel size 1mm, hit_ratio_thresh 4mm, 60% overlap. Increasing the RANSAC iteration number helps a lot but brings the computational overhead.

Great work and thank you for making it clear!

**Comment by chrischoy** [comment]
I thought about this problem of registering a complete CAD to a partial CAD model and this problem is fundamentally ill-posed if the CAD model has 2-way or 4-way symmetry.

I think one way to solve this problem would be limiting the random rotation augmentation to make the correspondences unique. Otherwise the hardest-negative mining will always mine the symmetric parts as the negative and screw up the training.
